### PR TITLE
Task/add range update callback

### PR DESCRIPTION
**Pull request by canavandl**
ref issue: #2346 

This PR adds callbacks to range objects that are fired on range changes.

todo:
- consider returning simpler cb_obj than "@model"
- requires better description of cb_obj in callback attr helpstring.
- probably style fixes

### CODE REVIEW

**Comment by canavandl** [comment]
I'm actually ok with this. It'll only trigger a callback on updates of the range containing the callback (wheel scrolls on x-axis won't trigger y-axis callbacks) and the cb_obj contains the range data under cb_obj.get('frame').get(*_range) (which seems reasonable).

**Comment by damianavila** [comment]
Seems OK to me, but pinging @bryevdv who made the callback feature and have more insight for sure...

**Comment by canavandl** [comment]
This should be able to be merged without the change to the callback interface (only the model gets passed to the callback)

**Comment by bryevdv** [comment]
Should include a section in the User's guide with a minimal example, that also explains this is only triggered by tools updating the ranges.

**Comment by damianavila** [comment]
@canavandl is the user guide minimal example here something you are working now? If you are not ready with the examples in the user guide we can merge this one before the freeze and add the docs later...

**Comment by canavandl** [comment]
@damianavila - I'm almost done w/ the example. I'll push it shortly

File: sphinx/source/docs/user_guide/interaction.rst [hunk 1]
```diff
@@ -353,6 +353,64 @@ hover tool is over.
 
     show(p)
 
+  Callbacks for Range Update
```
Comment (damianavila) [inline sphinx/source/docs/user_guide/interaction.rst]: I would add a

`.. _userguide_interaction_actions_range_callbacks:`

before the title to have it available to link it later...

File: sphinx/source/docs/user_guide/interaction.rst [hunk 2]
```diff
@@ -353,6 +353,64 @@ hover tool is over.
...
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (damianavila) [inline sphinx/source/docs/user_guide/interaction.rst]: Blank line after this...

File: sphinx/source/docs/user_guide/interaction.rst [hunk 3]
```diff
@@ -353,6 +353,64 @@ hover tool is over.
...
+  With Bokeh, ranges have a callback attribute that accept a Callback instance
+  will execute javascript code on range updates that are triggered by tool
```
Comment (damianavila) [inline sphinx/source/docs/user_guide/interaction.rst]: `It will`... I think

**Comment by canavandl** [comment]
I have an indentation problem and forgot to add the target name. Lemme fix

**Comment by damianavila** [comment]
> I have an indentation problem and forgot to add the target name. Lemme fix

OK, I made some two minor comments as well

**Comment by canavandl** [comment]
I see them. Building locally now

**Comment by damianavila** [comment]
Waiting for TravisCI, merging soon...

**Pull request merged**